crates/ln-dlc-node/src/util.rs

```diff
@@ -29,3 +33,52 @@ pub fn build_socket_address(ip: IpAddr, port: u16) -> SocketAddress {
 pub fn into_socket_addresses(address: SocketAddr) -> Vec<SocketAddress> {
     vec![build_socket_address(address.ip(), address.port())]
 }
+
+pub fn parse_from_uuid(uuid: Uuid) -> ReferenceId {
+    let hex = uuid.as_simple().to_hex();
+    let bytes = hex.as_bytes();
+
+    debug_assert!(bytes.len() == 32, "length must be exactly 32 bytes");
+
+    let mut array = [0u8; 32];
+    array.copy_from_slice(bytes);
+
+    array
+}
+
+pub fn parse_from_reference_id(reference_id: ReferenceId) -> Result<Uuid> {
+    let reference_id = hex::decode(reference_id)?;
+    let reference_id = reference_id.to_hex();
+    let reference_id = Uuid::parse_str(&reference_id)?;
```

Comment:
🔧 I think we should use `from_utf8` to go from `ReferenceId` to hex string. And then we can either parse the hex `Uuid` directly using `parse_str` or first convert to 16 bytes and then use `Uuid::from_slice` like I demonstrated above.

Probably better to keep it shorter, but the latter might be a bit clearer.